copy: fix nil pointer in COPY telemetry logging

rafiss opened the pull request:
fixes https://github.com/cockroachdb/cockroach/issues/102494

Release note (bug fix): Fixed a panic that could occur while a COPY statement is logged for telemetry purposes.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/103738)
<!-- Reviewable:end -->

otan commented on pkg/sql/conn_executor.go:
```diff
@@ -2945,12 +2945,10 @@ func (ex *connExecutor) execCopyIn(
 		initPlanner: func(ctx context.Context, p *planner) {
 			ex.initPlanner(ctx, p)
 		},
-		resetPlanner: func(ctx context.Context, p *planner, txn *kv.Txn, txnTS time.Time, stmtTS time.Time) {
```
should we delete this object from the struct too?

rafiss replied:
oops my fix is invalid. i forgot about the mode where we batch copies and call this `resetPlanner` in between each batch.

rafiss committed:
copy: fix nil pointer in COPY telemetry logging

Release note (bug fix): Fixed a panic that could occur while a
COPY statement is logged for telemetry purposes.

rafiss commented:
went with a more basic fix. please take another look!
